audioread/rawread.py

```diff
@@ -109,6 +109,9 @@ def read_data(self, block_samples=1024):
                 break
 
             # Make sure we have the desired bitdepth and endianness.
+            if old_width == 1:
```

Comment:
If I'm reading the audiooop docs correctly:

> In some audio formats, such as .WAV files, 16, 24 and 32 bit samples are signed, but 8 bit samples are unsigned.

we should only do this conversion when reading from WAV files (not via the sunau or aifc modules). So I think this line needs an extra check regarding the underlying audio format.